tests/phpunit/tests/html/wpHtmlTagProcessor.php

```diff
@@ -0,0 +1,1960 @@
+	 * @return array[].
+	 */
+	public function data_next_tag_ignores_script_tag_contents() {
+		return array(
+			'Simple script tag'                          => array(
+				'<script><span class="d-none d-md-inline">Back to notifications</span></script><div></div>',
+			),
+
+			'Simple uppercase script tag'                => array(
+				'<script><span class="d-none d-md-inline">Back to notifications</span></SCRIPT><div></div>',
+			),
+
+			'Script with a comment opener inside should end at the next script tag closer (dash dash escaped state)' => array(
+				'<script class="d-md-none"><!--</script><div></div>-->',
+			),
+
+			'Script with a comment opener and a script tag opener inside should end two script tag closer later (double escaped state)' => array(
+				'<script class="d-md-none"><!--<script><span1></script><span2></span2></script><div></div>-->',
+			),
+
+			'Double escaped script with a tricky opener' => array(
+				'<script class="d-md-none"><!--<script attr="</script>"></script>"><div></div>',
+			),
+
+			'Double escaped script with a tricky closer' => array(
+				'<script class="d-md-none"><!--<script><span></script attr="</script>"><div></div>',
+			),
+
+			'Double escaped, then escaped, then double escaped' => array(
+				'<script class="d-md-none"><!--<script></script><script></script><span></span></script><div></div>',
+			),
+
+			'Script with a commented a script tag opener inside should at the next tag closer (dash dash escaped state)' => array(
+				'<script class="d-md-none"><!--<script>--><span></script><div></div>-->',
+			),
+
+			'Script closer with another script tag in closer attributes' => array(
+				'<script><span class="d-none d-md-inline">Back to notifications</title</span></script <script><div></div>',
+			),
+
+			'Script closer with attributes'              => array(
+				'<script class="d-md-none"><span class="d-none d-md-inline">Back to notifications</span></script id="test"><div></div>',
+			),
+
+			'Script opener with title closer inside'     => array(
+				'<script class="d-md-none"></title></script><div></div>',
+			),
+
+			'Complex script with many parsing states'    => array(
+				'<script class="d-md-none"><!--<script>--><scRipt><span><!--<span><Script</script>--></scripT><div></div>-->',
+			),
+		);
+	}
+
+	/**
+	 * @ticket 56299
+	 *
+	 * @covers WP_HTML_Tag_Processor::next_tag
+	 *
+	 * @dataProvider data_next_tag_ignores_contents_of_rcdata_tag
+	 *
+	 * @param string $rcdata_then_div HTML with RCDATA before a DIV.
+	 * @param string $rcdata_tag      RCDATA tag.
+	 */
+	public function test_next_tag_ignores_contents_of_rcdata_tag( $rcdata_then_div, $rcdata_tag ) {
+		$p = new WP_HTML_Tag_Processor( $rcdata_then_div );
+		$p->next_tag();
+		$this->assertSame( $rcdata_tag, $p->get_tag(), "The first found tag was not '$rcdata_tag'" );
+		$p->next_tag();
+		$this->assertSame( 'DIV', $p->get_tag(), "The second found tag was not 'div'" );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
```

Comment:
```suggestion
	 * @return array[] Test cases for the test_next_tag_ignores_contents_of_rcdata_tag method.
```

Reply:
@adamziel these were removed previously with a "for now" approach that is being used in Core. See https://github.com/WordPress/wordpress-develop/pull/3920#discussion_r1094585197. I hope you don't mind, but I'll close this as resolved.